### PR TITLE
ISSUE #2639 - Updated node version to 14.0.0

### LINKED ISSUE #2639: Update node version to 14.0.0
### Description
Update node version to 14.0.0 for being up to date.

### Goals
- [x] Update node version to 14.0.0 

### Tasks
- [x] Update node version
- [x] Udate packages that are needed to be updated for 14.0
- [x] Make sure everything works on backend
- [x] Make sure everything works on frontend

### PR DESCRIPTION
**Pull request by sanmont3drepo**
This fixes #2639 

#### Description
- Updated incompatibles packages for 14.0.0
- Min changes in logger

### CODE REVIEW

**Comment by keifgwinn** [comment]
Could we use 14.17.0 instead?

**Comment by carmenfan** [comment]
> Could we use 14.17.0 instead? There's lots of security fixes in since 14.0 https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V14.md#14.17.0

package.json doesn't specify, meaning it should let you run as long as major version is 14.

image

**Comment by keifgwinn** [comment]
> package.json doesn't specify, meaning it should let you run as long as major version is 14.

We are specifying 14.0.0 in travis however for tests, so should bump that up to at least current release on 14.

File: .travis.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ matrix:
       submodules: false
       depth: 1
       node_js:
-        - "10.15.1"
+        - "14.0.0"
```
Comment (carmenfan) [inline .travis.yml:10]: should use LTS version

File: .travis.yml [hunk 2]
```diff
@@ -23,7 +23,7 @@ matrix:
...
```
Comment (carmenfan) [inline .travis.yml:26]: same here

File: backend/logger.js [hunk 3]
```diff
@@ -128,12 +119,20 @@ repoLogger.prototype.logMessage = function (type, msg, meta) {
...
 	const metadata = Object.assign({}, meta);
 
-	this.session && this.session.user && (metadata.username = this.session.user.username);
-	this.req && this.req.method && (metadata.method = this.req.method);
-	this.req && this.req.originalUrl && (metadata.url = this.req.originalUrl);
+	if (this.session && this.session.user) {
+		metadata.username = this.session.user.username;
+	}
+
+	if (this.req && this.req.method) {
```
Comment (carmenfan) [inline backend/logger.js:126]: merge the 2 ifs (kind of) so we only test this.req once
```suggestion
	if (this.req) {
	    if (this.req.method) {
	        //...
	    }
	
	}
```

**Review by carmenfan — CHANGES_REQUESTED** [review]

**Review by carmenfan — APPROVED** [review]

**Pull request merged**